fiasco/ions.py

```diff
@@ -1576,48 +1576,40 @@ def two_photon(self, wavelength: u.angstrom,
         include_protons : `bool`, optional
             If True, use proton excitation and de-excitation rates in the level population calculation.
         """
-
         wavelength = np.atleast_1d(wavelength)
-        temperature = np.atleast_1d(self.temperature)
         electron_density = np.atleast_1d(electron_density)
 
-        prefactor = (const.h * const.c)
-
-        if not self.hydrogenic and not self.helium_like:
-            return u.Quantity(np.zeros(wavelength.shape + self.temperature.shape + electron_density.shape),
-                              'erg cm^3 s^-1 Angstrom^-1')
+        final_shape = wavelength.shape + self.temperature.shape + electron_density.shape
         if self.hydrogenic:
             A_ji = self._hseq['A']
             psi_norm = self._hseq['psi_norm']
             cubic_spline = CubicSpline(self._hseq['y'], self._hseq['psi'])
-            # Get the index of the 2S1/2 state for H-like:
-            level_index = np.where((self._elvlc['config'] == '2s') & (self._elvlc['J'] == 0.5))[0][0]
-        if self.helium_like:
+            config = '2s'  # Get the index of the 2S1/2 state for H-like
+            J = 0.5
+        elif self.helium_like:
             A_ji = self._heseq['A']
             psi_norm = 1.0 * u.dimensionless_unscaled
             cubic_spline = CubicSpline(self._heseq['y'], self._heseq['psi'])
-            # Get the index of the 1s2s 1S0 state for He-like:
-            level_index = np.where((self._elvlc['config'] == '1s.2s') & (self._elvlc['J'] == 0))[0][0]
+            config = '1s.2s'  # Get the index of the 1s2s 1S0 state for He-like:
+            J = 0
+        else:
+            return u.Quantity(np.zeros(final_shape),  'erg cm^3 s^-1 Angstrom^-1')
+        level_index = np.where((self._elvlc['config'] == config) & (self._elvlc['J'] == J))[0][0]
 
-        # store the rest wavelength:
         E_obs = self._elvlc['E_obs'][level_index]
         E_th = self._elvlc['E_th'][level_index]
         E_2p = E_obs if E_obs > 0.0 else E_th
-        rest_wavelength = 1 / (E_2p.to(1/u.angstrom))
+        rest_wavelength = 1 / E_2p
 
-        psi_interp = cubic_spline(rest_wavelength / wavelength)
-        indices = np.where(wavelength < rest_wavelength)
-        psi_interp[indices] = 0.0  # intensity below rest_wavelength is 0
+        psi_interp = cubic_spline((rest_wavelength / wavelength).decompose())
+        psi_interp = np.where(wavelength < rest_wavelength, 0.0, psi_interp)
 
         energy_dist = (A_ji * rest_wavelength * psi_interp) / (psi_norm * wavelength**3)
 
-        level_population = self.level_populations(electron_density,
-                                                  include_protons=include_protons)[:,:,level_index]
+        level_population = self.level_populations(electron_density, include_protons=include_protons)
+        level_population = level_population[..., level_index]
```

Comment:
Why the change in syntax here?

Reply:
The `...` allow any number of dimensions to come before the last index so if `level_population` ever becomes anything but three dimensional, this still works, provided the last dimension indexes the energy levels.